Change docs related to `show-backup` command new output (#990)
EnterpriseDB/barman: doc/barman.1.d/50-show-backup.md

show-backup *SERVER_NAME* *BACKUP_ID*
:   Show detailed information about a particular backup, identified by
    the server name and the backup ID. See the [Backup ID shortcuts](#shortcuts)
    section below for available shortcuts. The following example is from

>     an incremental backup (which requires Postgres version >= 17):

Reviewer: ```suggestion
    a block-level incremental backup (which requires Postgres version >= 17):
```

Just so we use a similar term to the one I saw in another PR.

    Retention Policy     : not enforced
    Previous Backup      : 20150821T130001
    Next Backup          : - (this is the latest base backup)
    Root Backup          : 20240814T015504
    Parent Backup        : 20240814T016504
    Backup chain size    : 3
    Children Backup(s)   : 20240814T018515
```

> **NOTE:**
> Depending on the version of your Postgres Server or the type
> of the backup, the output may vary. For example, "incremental" backups

> > were introduced in Postgres 17, so fields like "WAL summarizer",

Reviewer 2: ```suggestion
> were introduced in Postgres 17, so incremental backups specific fields like "WAL summarizer",
```

> "Root Backup", "Parent Backup", "Backup chain size", "Children Backup(s)"

> > will not be present in outputs with a previous version.

Reviewer 2: ```suggestion
> will not be present in the output of the previous Postgres versions.
```

>

> > Below you can find the fields that are specific by type:

Reviewer 2: ```suggestion
> Below you can find a list of type-specific fields:
```

>
> `Resource savings`: "rsync" and "incremental"
> `Root Backup`, `Parent Backup`, `Backup chain size`: "incremental"

> > `Children Backup(s)`: "full" and "incremental"

Reviewer 2: ```suggestion
> `Children Backup(s)`: available for "full" and "incremental" backups
```

>
> Also, if a backup was created using a barman release < 3.11, there
> might be other fields omitted in the output, such as `WAL summarizer`

> > and the fields related to "incremental" backups.

Reviewer 2: mmm is not entirely clear, please reword @andremagui

    Retention Policy     : not enforced
    Previous Backup      : 20150821T130001
    Next Backup          : - (this is the latest base backup)
    Root Backup          : 20240814T015504
    Parent Backup        : 20240814T016504
    Backup chain size    : 3
    Children Backup(s)   : 20240814T018515
```

> **NOTE:**
> Depending on the version of your Postgres Server or the type
> of the backup, the output may vary. For example, "incremental" backups
> were introduced in Postgres 17, so incremental backups specific fields
> like "WAL summarizer", "Root Backup", "Parent Backup", "Backup chain size",
> "Children Backup(s)" will not be present in the output of the previous

> > Postgres versions.

Author: I think before barman 3.11, this was already the way things worked, all "postgres" being full backups. I think it looks confusing the way it is written. I would try to be less wordy, just adding an important bit. Something like 
`For example, "incremental" backups were introduced in Postgres 17, so incremental backups specific fields like "WAL summarizer", "Root Backup", "Parent Backup", "Backup chain size", "Children Backup(s)" will not be present in the output of the previous Postgres versions **nor will the backup_type="incremental" exist.**`

But i really don't know if how we should add this info here? Thoughts? @gustabowill @barthisrael @gcalacoci

> > Postgres versions.

Reviewer: Maybe we could write this way instead:

```md
> **NOTE:**
> Depending on the version of your Postgres Server and/or the type
> of the backup, the output of `barman show-backup` command may
> be different. For example, fields like "Root Backup", "Parent Backup",
> "Backup chain size", and "Children Backup(s)" only make sense when
> showing information about a block-level incremental backup taken
> with `backup_method = postgres` and using Postgres 17 or newer,
> thus those fields are omitted for other kind of backups or older versions
> of Postgres.
```

>

> > Below you can find a list of type-specific fields:
> >
> > `Resource savings`: available for "rsync" and "incremental" backups
> > `Root Backup`, `Parent Backup`, `Backup chain size`: available for 
> > "incremental" backups only
> > `Children Backup(s)`: available for "full" and "incremental" backups

Reviewer: ```suggestion
> These are the possible values for the field "Backup Type":
>
> * `rsync`: for a backup taken with `rsync`;
> * `full`: for a full backup taken with `pg_basebackup`;
> * `incremental`: for an incremental backup taken with `pg_basebackup`;
> * `snapshot`: for a snapshot-based backup taken in the cloud.
>
> Below you can find a list of fields that may be shown or omitted depending
> on the type of the backup:
>
> `Resource savings`: available for "rsync" and "incremental" backups;
> `Root Backup`, `Parent Backup`, `Backup chain size`: available for 
> "incremental" backups only;
> `Children Backup(s)`: available for "full" and "incremental" backups;
> `Snapshot information`: available for "snapshot" backups only.
```

>

> > Please note that if a backup was created with Barman version 3.10 or

Reviewer 2: in this paragraph you are sorta repeating what you said in the first one @andremagui

Author: Yeah, but here we need to document the fact about the PostgreSQL version and the Barman version on the output.

Any suggestions on how to organize so we do not repeat documentation here?
@barthisrael @gcalacoci @bleidens @gustabowill

Author: is this understandable?
> **NOTE:**
> With the introduction of "incremental" backups in PostgreSQL 17 and in
> Barman 3.11, depending on the version of your Postgres Server, the 
> Barman version and the type of the backup, the output may vary.
>
> Below you can find a list of type-specific fields:
>
> `Resource savings`: available for "rsync" and "incremental" backups
> `Root Backup`, `Parent Backup`, `Backup chain size`: available for 
> "incremental" backups only
> `Children Backup(s)`: available for "full" and "incremental" backups
>
> Please note that if a backup was created with Barman version 3.10 or
> earlier, and you use the `show-backup` command with Barman version 3.11
> or later, the output may not include certain fields. This is also true
> when using the command on a backup created from a Postgres server 
> with version <= 16. Specifically, fields such as `WAL summarizer` and 
> those related to "incremental" backups may be missing from the output
> because they were introduced in version 3.11.

> earlier, and you use the `show-backup` command with Barman version 3.11
> or later, the output may not include certain fields. Specifically, 
> fields such as `WAL summarizer` and those related to "incremental" 
> backups may be missing from the output because they were introduced 

> > in version 3.11.

Reviewer: This is part of a `man` page.

I don't think this will render nicely through `man`.

Author: Indeed, i will fix!

    Retention Policy     : not enforced
    Previous Backup      : 20150821T130001
    Next Backup          : - (this is the latest base backup)
    Root Backup          : 20240814T015504
    Parent Backup        : 20240814T016504
    Backup chain size    : 3
    Children Backup(s)   : 20240814T018515
```

> 

Author: Please, this is looking promising Please, take another look for feedback/suggestions! @gcalacoci @barthisrael @bleidens @gustabowill

Reviewer 2: tbh everything LGTM
